tests/Maintenance/MaintenanceTest.cpp

```diff
@@ -0,0 +1,527 @@
+      props->add("hello", VPackValue("world")); }
+    ActionDescription desc({{"name", "SomeAction"}}, props);
+    REQUIRE(desc.get("name") == "SomeAction");
+    REQUIRE(desc.properties()->slice().hasKey("hello"));
+    REQUIRE(desc.properties()->slice().get("hello").copyString() == "world");
+  }
+    
+  SECTION("Retrieve double value from ActionDescription's properties") {
+    double pi = 3.14159265359;
+    auto props = std::make_shared<VPackBuilder>();
+    { VPackObjectBuilder obj(props.get());
+      props->add("pi", VPackValue(3.14159265359)); }
+    ActionDescription desc({{"name", "SomeAction"}}, props);
+    REQUIRE(desc.get("name") == "SomeAction");
+    REQUIRE(desc.properties()->slice().hasKey("pi"));
+    REQUIRE(
+      desc.properties()->slice().get("pi").getNumber<double>() == pi);
+  }
+    
+  SECTION("Retrieve integer value from ActionDescription's properties") {
+    size_t one = 1;
+    auto props = std::make_shared<VPackBuilder>();
+    { VPackObjectBuilder obj(props.get());
+      props->add("one", VPackValue(one)); }
+    ActionDescription desc({{"name", "SomeAction"}}, props);
+    REQUIRE(desc.get("name") == "SomeAction");
+    REQUIRE(desc.properties()->slice().hasKey("one"));
+    REQUIRE(
+      desc.properties()->slice().get("one").getNumber<size_t>() == one);
+  }
+    
+  SECTION("Retrieve array value from ActionDescription's properties") {
+    double pi = 3.14159265359;
+    size_t one = 1;
+    std::string hello("hello world!");
+    auto props = std::make_shared<VPackBuilder>();
+    { VPackObjectBuilder obj(props.get());
+      props->add(VPackValue("array"));
+      { VPackArrayBuilder arr(props.get());
+        props->add(VPackValue(pi));
+        props->add(VPackValue(one));
+        props->add(VPackValue(hello)); }}
+    ActionDescription desc({{"name", "SomeAction"}}, props);
+    REQUIRE(desc.get("name") == "SomeAction");
+    REQUIRE(desc.properties()->slice().hasKey("array"));
+    REQUIRE(desc.properties()->slice().get("array").isArray());
+    REQUIRE(desc.properties()->slice().get("array").length() == 3);
+    REQUIRE(desc.properties()->slice().get("array")[0].getNumber<double>()==pi);
+    REQUIRE(desc.properties()->slice().get("array")[1].getNumber<size_t>()==one);
+    REQUIRE(desc.properties()->slice().get("array")[2].copyString() == hello );
+  }
+    
+}
+
+TEST_CASE("ActionPhases", "[cluster][maintenance]") {
+
+  auto plan = createNode(planStr);
+  auto current = createNode(currentStr);
+
+  std::vector<std::string> dbsIds;
+  for (auto const& dbs : plan("Plan/DBServers").children()) {
+    dbsIds.push_back(dbs.first);
+  }
+  
+  std::map<std::string, Node> localNodes {
+    {dbsIds[2], createNode(dbs1Str)},
+    {dbsIds[0], createNode(dbs2Str)},
+    {dbsIds[1], createNode(dbs3Str)}};  
+
+  SECTION("In sync should have 0 effects") {
+    
+    std::vector<ActionDescription> actions;
+    
+    for (auto const& node : localNodes) {
+      arangodb::maintenance::diffPlanLocal(
```

Comment:
I get an exception thrown here and 10 more similar locations afterward.  line 286 of Maintenance.cpp is upset that it gets a null/empty slice (or such).